### PR TITLE
Remove filtering test cases by test suite ID in TestCase list API

### PR DESCRIPTION
**Pull request by sureshms**
### Describe your changes :

TestSuite GET API already has a way to get test cases for the given test suite ID. Removing the redundant filter functionality on TestCase list API.

#
### Type of change :

- [x] Improvement

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshach — DISMISSED** [review]

**Comment by harshach** [comment]
@sureshms lets hold off merging this. we may have dependencies

File: ingestion/src/metadata/test_suite/api/workflow.py [hunk 1]
```diff
@@ -287,27 +287,6 @@ def get_or_create_test_suite_entity_for_cli_workflow(
...
-        for test_suite in test_suites:
-            test_case_entity_list = self.metadata.list_entities(
-                entity=TestCase,
-                fields=["testSuite", "entityLink", "testDefinition"],
-                params={"testSuiteId": test_suite.id.__root__},
-            )
-            test_cases_entity.extend(test_case_entity_list.entities)
-
-        return test_cases_entity
-
```
Comment (TeddyCr) [inline ingestion/src/metadata/test_suite/api/workflow.py:310]: This will break the ingestion workflow

**Comment by harshach** [comment]
Lets handle this API post release for 0.12.1

**Comment by harshach** [comment]
@TeddyCr @ayush-shah can we incorporate this for 0.13.1? 
lets make sure we follow - up in next release

**Pull request closed without merging**